Classes/Event/EventPublisher.php

```diff
@@ -105,7 +106,10 @@ public function publishMany(string $streamName, array $events, int $expectedVers
         foreach ($rawEvents as $rawEvent) {
             $eventClassName = $this->eventTypeResolver->getEventClassNameByType($rawEvent->getType());
             $event = $this->propertyMapper->convert($rawEvent->getPayload(), $eventClassName, $configuration);
-            foreach ($this->eventListenerLocator->getSynchronousListeners($rawEvent->getType()) as $listener) {
+            foreach ($this->eventListenerLocator->getSynchronousListenersByEventType($rawEvent->getType()) as $listener) {
+                if ($listener[0] instanceof ActsBeforeInvokingEventListenerMethodsInterface) {
+                    $listener[0]->beforeInvokingEventListenerMethod($listener[1], $event, $rawEvent);
```

Comment:
`$listener` is still not an array (I keep repeating myself in case you didn't see it - as you didn't respond yet).
With the current implementation it *is* an array, but the signature says `\callable` and the idea was that you could have static or closure listeners.. It might be a case of YAGNI, but maybe we should at least add a comment here?